Mage/src/main/java/mage/abilities/TriggeredAbilityImpl.java

```diff
@@ -236,7 +236,11 @@ public String getRule() {
                 sb.append(" Do this only once each turn.");
             }
         }
-        return sb.toString();
+        String rule = sb.toString();
+        if (nameReplacement != null) {
+            rule = rule.replace("{this}", nameReplacement);
```

Comment:
Looks bad. You need manually change each xmage's ability to support new permanent names (1200+ overrides of `getRule`). Impossible.

I recommends to manually set/generate rules for Case related abilities (you can override `getRule`, generate text and replace "this" by same way).

Not related to that PR but can be useful for @xenohedron's text gen refactors:

If you want to implement name replacement logic then it must be a global implementation for "this" replacement code. Example: search code by `"{this}", ` (there are not so many places to fix). Also it must support planeswalker names (diff names for diff "this" index, e.g. first for short name, full for second name, etc) and spell short names (I don't remember examples but they are introduced in last 2 years).

![shot_240128_171921](https://github.com/magefree/mage/assets/8344157/d1362c0f-d449-4ae2-aafb-f21d650e0316)

Possible solution 1:
* `MageObject::setName(fullName)/(fullName, shortName)` or `MageObject::setAdditionalNames(shortName)`
* Shared `CardUtil::replaceName(text, object)/(text, objectName)` for name replacement code;

Possible solution 2:
* Solution 1 + introduce "{this}" and "{this.short}" for the rules.

Reply:
I have removed the general text changes and moved them into the specific card implementations for now.